Sources/NIO/NIOContiguousCollection.swift

```diff
@@ -13,7 +13,7 @@
 //===----------------------------------------------------------------------===//
 
 /// A `Collection` that is contiguously laid out in memory and can therefore be duplicated using `memcpy`.
-public protocol ContiguousCollection: Collection {
+public protocol NIOContiguousCollection: Collection {
     @inlinable
     func withUnsafeBytes<R>(_ body: (UnsafeRawBufferPointer) throws -> R) rethrows -> R
```

Comment:
We should namespace this function too.

Reply:
yes, thanks now done

Reply:
@Lukasa I think we should just close this and take #709 . @itaiferber's suggestion is much better, we don't need `ContiguousCollection` anymore at all...

Reply:
Also, in case it helps, Foundation now offers the `ContiguousBytes` protocol if that also suits your needs.
(I don't have a particular stake in this; either way, we just need to ensure that SR-9505 is resolved.)